### PR TITLE
Add documentation on using remote URIs for runtime environments

### LINKED ISSUE #20340: [runtime_env] Document remote URI support
- Specify you can pass remote URIs for working_dir, py_modules
- Add a section describing the options supported (s3, gs, http) and pre-reqs for them (e.g., pip installs)
- Short "howto" on how to use HTTP for git repos

### PR DESCRIPTION
**Pull request by shrekris-anyscale**
## Why are these changes needed?

Feature #19862 allows users to specify dependencies accessed via remote URIs in their `runtime_env` dictionaries. These changes update the ``runtime_env`` documentation to teach users how to use remote URIs.

## Related issue number

Closes #20340.

## Checks

- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [X] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - Documentation PR– no testing needed.

### CODE REVIEW

File: doc/source/dependency-management.rst [hunk 1]
```diff
@@ -125,3 +135,143 @@ Here are some examples of runtime environments combining multiple options:
...
+Suppose you upload the compressed ``example_dir`` directory to AWS S3 at the S3 URI ``s3://example_bucket/example.zip``.
+Your ``runtime_env`` dictionary should contain:
+
+.. code-block:: python
+
+    runtime_env = {..., "working_dir": "s3://example_bucket/example.zip", ...}
+
+.. warning::
+
+  Check for hidden files in zipped dependencies.
```
Comment (architkulkarni) [inline doc/source/dependency-management.rst:171]: Can you give a brief explanation or example for how to do this?

File: doc/source/dependency-management.rst [hunk 2]
```diff
@@ -125,3 +135,143 @@ Here are some examples of runtime environments combining multiple options:
...
+The contents of this directory will be directly accessed as the ``working_dir`` or a ``py_module``.
+
+For example, suppose you want to use the contents in your local ``/some_path/example_dir`` directory as your ``working_dir``.
+If you want to specify this directory as a local path, your ``runtime_env`` dictionary should contain:
+
+.. code-block:: python
+
+    runtime_env = {..., "working_dir": "/some_path/example_dir", ...}
+
+Suppose instead you want to host its contents remotely and provide a remote URI.
```
Comment (architkulkarni) [inline doc/source/dependency-management.rst:157]:
```suggestion
Suppose instead you want to host your files remotely and provide a remote URI.
```

File: doc/source/dependency-management.rst [hunk 3]
```diff
@@ -125,3 +135,143 @@ Here are some examples of runtime environments combining multiple options:
...
+You would need to first compress the ``example_dir`` directory into a zip file.
+There should be no other files or directories at the top level of the zip file, other than ``example_dir``.
```
Comment (architkulkarni) [inline doc/source/dependency-management.rst:159]: Is it possible to give a quick example of how to do this, like an example command or something?

**Review by architkulkarni — APPROVED** [review]
This looks great!  We'll use this as a model for expanding on the docs for the other runtime_env fields like `conda`.

**Comment by javi-redondo** [comment]
My main feedback is for the URLs, it seems easier to tell people where to click/what to run to get the URLs rather than the structure of the URL (we can show both, but it's easier to point to how to get it rather than how to construct it). Otherwise lgtm!

**Comment by architkulkarni** [comment]
Can we rearrange things to the recommended link option appears before the non-recommended one?  And within the recommended option, using `HEAD` is not recommended, so perhaps it should go last.  Other than that, looks good!

**Review by architkulkarni — APPROVED** [review]

**Review by edoakes — APPROVED** [review]

**Pull request merged**